recipes/mbedtls/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+from conans import CMake, ConanFile, tools
+import os
+
+
+class MBedTLSConan(ConanFile):
+    name = "mbedtls"
+    description = "mbed TLS makes it trivially easy for developers to include cryptographic and SSL/TLS capabilities in their (embedded) products"
+    topics = ("conan", "mbedtls", "polarssl", "tls", "security")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://tls.mbed.org"
+    license = ("GPL-2.0", "Apache-2.0",)
+    exports_sources = "CMakeLists.txt",
+    generators = "cmake",
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_zlib": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_zlib": True,
+    }
+    _source_subfolder = "source_subfolder"
+
+    @property
+    def _version(self):
+        return self.version.rsplit("-", 1)[0]
+
+    @property
+    def _license(self):
+        return self.version.rsplit("-", 1)[1]
+
+    def config_options(self):
+        if self.settings.os == "Windows" or self.options.shared:
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+    def requirements(self):
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "{}-{}".format(self.name, self._version)
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _get_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["USE_SHARED_MBEDTLS_LIBRARY"] = self.options.shared
+        cmake.definitions["USE_STATIC_MBEDTLS_LIBRARY"] = not self.options.shared
+        cmake.definitions["ENABLE_ZLIB_SUPPORT"] = self.options.with_zlib
+        cmake.definitions["ENABLE_PROGRAMS"] = False
+        cmake.definitions["ENABLE_TESTING"] = False
+        return cmake
+
+    def _patch_sources(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "library", "CMakeLists.txt"),
+                              "DESTINATION ${LIB_INSTALL_DIR}",
+                              "ARCHIVE DESTINATION ${LIB_INSTALL_DIR} LIBRARY DESTINATION {LIB_INSTALL_DIR} RUNTIME DESTINATION bin")
+        x509_crt_h = os.path.join(self._source_subfolder, "include", "mbedtls", "x509_crt.h")
+        tools.replace_in_file(x509_crt_h,
```

Comment:
extract to patch file, if possible

Reply:
I've extracted them to patches. One for apache and gpl though.
Because `tools.patch` [does not support fuzzy patching (yet)](https://github.com/conan-io/conan/issues/2752).